solutions/fully-configurable/main.tf

```diff
@@ -0,0 +1,392 @@
+########################################################################################################################
+# Resource Group
+########################################################################################################################
+
+# Input variable validation
+locals {
+  prefix = var.prefix != null ? (var.prefix != "" ? var.prefix : null) : null
+}
+
+module "resource_group" {
+  count                        = 1
+  source                       = "terraform-ibm-modules/resource-group/ibm"
+  version                      = "1.1.6"
+  existing_resource_group_name = var.existing_resource_group_name
+}
+
+#######################################################################################################################
+# KMS keys
+#######################################################################################################################
+
+# parse KMS details from the existing KMS instance CRN
+module "existing_kms_instance_crn_parser" {
+  count   = var.key_management_service_encryption_enabled && var.existing_key_management_service_instance_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_key_management_service_instance_crn
+}
+
+# If existing KMS root key CRN passed, parse details from it
+module "kms_root_key_crn_parser" {
+  count   = var.existing_key_management_service_root_key_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_key_management_service_root_key_crn
+}
+
+locals {
```

Comment:
All this validation need to move to terraform 1.9 cross variable validation in variables.tf